docs/assignment3.md

```diff
@@ -1 +1,205 @@
+* Updated class diagram to properly represent JSON objects 
+* Updated object diagram to represent the completed implementation and any changes that were made from the assignment 2 code
+* Added association names to the class diagram 
+* Added the chat room feature (Added **ChatClient** class) so that you can communicate with other players (use command "chat")
+* Implemented time limit feature to make the game slightly more complex
+* Added obstacles to the game using the **Obstacle** class to make the game more complex
+* Added the "use" command (implemented as the **Use** class) to the game to apply **Item** to **Obstacles** to clear them
+* Implemented persistence to allow a user to save their progress. The objective behind this feature is to allow users to begin a new game, play, exit the terminal, and then restart in the same place next time they open it. This feature gives users the choice of whether or not to save their current game state, including the items they hold, their current location, and their time within the time limit, and pick up where they left off the next time they want to play. 
+
+### Application of design patterns
+Author(s): Sarah
+
+![Class Diagram with design patterns labelled](https://github.com/sarahpazik/software-design-vu/blob/Assignment3/Copy%20of%20class%20diagram.png)
+
+| ID  | DP1  |
+|---|---|
+| **Design pattern**  | Decorator |
+| **Problem**  | We wanted a way to give the player the ability to get rid of the time limit in the game, but had no easy way to do so. |
+| **Solution**  | The decorator pattern allowed us to have a new decorated player so that when the player finds the broken clock in the game, they are turned into a decorated player who no longer has a time limit. |
+| **Intended use**  | We have a "broken clock" item in each game, and in our "pick" class/command, it checks whether the item picked up was the broken clock. If it was, the player's reference is changed to a TimeDecoratedPlayer rather than a RegularPlayer. This makes it so that the player's checkTime() method no longer counts down the time. |
+| **Constraints**  | N/A |
+
+| ID  | DP2  |
+|---|---|
+| **Design pattern**  | Command |
+| **Problem**  | We wanted a way to firstly separate our commands, since our Action class was getting complicated and hard to follow, and secondly be able to store the user's commands. |
+| **Solution**  | This design pattern allowed us to split each of the commands into separate classes, solving our first problem. It also allows us to store our commands like any other objects, solving the second problem. |
+| **Intended use**  | Now, when a user types a command, it calls the respective class's execute() method, which overrides the Command interface's execute() method. Now, each command can be stored as a separate object rather than just existing as a method in the Action class.  |
+| **Constraints**  | N/A |
+
+Maximum number of words for this section: 2000
+
```

Comment:
-Well motivated use of the command pattern to solve key issues with the project. Great!
-The decorator is used in a limited, niche way. What about additional gameplay alterations akin to removing the time limit? The problem statement is slightly off, you usually wouldn't use a design pattern to easily remove something, rather you would be interested in the added behavior that is provided from not having a time-limit i.e less stressful exploration.